internal/controller/rbac/provider/roles/reconciler.go

```diff
@@ -444,3 +450,17 @@ func (d OrgDiffer) Differs(a, b string) bool {
 
 	return oa != ob
 }
+
+// isLocalBuild checks if the reference is a local build without registry info
+func isLocalBuild(ref string) bool {
+	return !strings.Contains(ref, "/")
+}
+
+// extractLocalOrg extracts the organization part from a local build reference
+func extractLocalOrg(ref string) string {
+	parts := strings.Split(ref, "-")
+	if len(parts) > 2 {
+		return parts[1]
+	}
+	return ref
+}
```

Comment:
Is this style of local build common, or just something our build submodule does? Does this change help anything other than our CI system?

I'd rather not add code to special case for things our build system does. Could we update the build system instead?

Reply:
wee have support for install packages from local cache folder - but it's not working for family providers for RBAC to add providerconfig rights for example- last possible xp version was v1.14.x which was working

Reply:
and i don't like the facts that we pin the XP version to v1.14.x in provider repos

Reply:
A couple thoughts/observations:

* I tested these unit test additions (just the unit tests, not any of @haarchri's new logic) on the [`release-1.14` branch](https://github.com/crossplane/crossplane/tree/release-1.14) and the `SameLocalOrg` case fails there as well.
* The only differences in v1.15 compared to v1.14 for anything under [`internal/controller/rbac`](https://github.com/crossplane/crossplane/tree/release-1.15/internal/controller/rbac) are just for a single unit test case added - no logic was changed in v1.15 under this directory, which is also what I see in the PR that changed the default registry https://github.com/crossplane/crossplane/pull/5261. Diff of that dir between v1.14 and v1.15:
```
❯ git --no-pager diff upstream/release-1.14:internal/controller/rbac upstream/release-1.15:internal/controller/rbac
diff --git a/provider/roles/reconciler_test.go b/provider/roles/reconciler_test.go
index 6c1209dbe..80c530e26 100644
--- a/provider/roles/reconciler_test.go
+++ b/provider/roles/reconciler_test.go
@@ -458,6 +458,12 @@ func TestOrgDiffer(t *testing.T) {
 			b:        "index.docker.io/cool/other-provider:v1.0.0",
 			want:     true,
 		},
+		"DifferentRegistriesWithDefaulting": {
+			registry: "xpkg.example.org",
+			a:        "index.docker.io/cool/provider:v1.0.0",
+			b:        "cool/other-provider:v1.0.0",
+			want:     true,
+		},
 	}

 	for name, tc := range cases {
```

That all implies v1.15 should work the same as v1.14 for this scenario, but @haarchri is reporting that it is not working the same, so I'm a bit puzzled by that. Is it possible that we've also changed the way the local build creates/names packages in the meantime? 🤔 

Either way, I can see the scenario that @haarchri is trying to address, but I would suggest approaching it in a slightly different way that is more aligned with OCI conventions.  In the current PR, it looks like we are parsing references in a manner specific to how builds are naming packages with `-` in the names.

Instead, perhaps we should handle this as the case where the namespace component of a package is unspecified, e.g. a missing namespace from `registry/namespace/repository:tag`

Namespaces/Orgs are optional as far as I can tell, e.g. for dockerhub if you [don't specify a namespace then it assumes `library`](https://docs.docker.com/guides/docker-concepts/building-images/build-tag-and-publish-an-image/#tagging-images), but I'm having trouble finding that in the official OCI specs.

It seems reasonable to me that if the registries match and both images are not specifying a namespace/org, then that should be counted as a match too, e.g. the orgs do not differ. They should both be using the same resulting namespace/org on their registry of choice.

Can we take that approach for this implementation instead of having specific knowledge of how the package names are formed from our local builds? It feels more aligned with the general OCI ecosystem approaches.

Reply:
You can see here https://github.com/crossplane-contrib/provider-upjet-aws/pull/1330 1.14.x is working but no 1.15.x and no 1.16.x and it is the same naming schema